### DISCUSSION TITLE
Undefined Type for Operation

### DISCUSSION THREAD

**Discussion by Undistraction** [urql-graphql/urql#1160] [Q&A] [answered]
I've been using URQL for a while without issues, but I'm in the process of integrating the `authExchange`, based on the [example](https://github.com/FormidableLabs/urql/tree/main/exchanges/auth#quick-start-guide) provided in the documentation. I'm hitting a problem where it appears a query to `currentUser` is being dropped. I can't see any request made to the server, and `useQuery` enters and remains in a fetching state. 

I'm getting a warning in the console when making the query: 
> No exchange has handled operations of type "undefined". Check whether you've added an exchange responsible for these operations. 

And if I check the operation from with the auth exchange's `addAuthToOperation` callback, I can see that this is true. Both `kind` and `operationName` are `undefined`. 

I'm assuming that this is the cause of the problem, but what could cause the operation type to be `undefined`?

My config looks like this:

```
createClient({
    url,
    fetchOptions: {
      mode: `cors`,
      credentials: `include`,
    },
    exchanges: [
      devtoolsExchange,
      dedupExchange,
      cacheExchange({ ...(schema && { schema }) }),
      errorExchange({onError: (error ) => {
        const { request, response, graphqlErrors } = error
        errorReporter.send(
          error,
          { from: `api`, type: `graphql` },
          {
            graphqlErrors,
            request: prettyPrintObj(request),
            response: prettyPrintObj(response),
          }
        )
      }})
      authExchange({
        ...getAuthExchangeConfig({ localStorageApi, authApi, authorize }),
      }),
      fetchExchange,
    ],
  })
```

Thanks.

**Comment by kitten** [comment, marked as answer]
It's possible that this was caused by `operation.kind`, which is a new addition from [`@urql/core@1.14.0`](https://github.com/FormidableLabs/urql/blob/main/packages/core/CHANGELOG.md#1140).

Now, there's a simple fix for this. If you're using `urql` for instance, bump that, or bump just `@urql/core`. You'll have to make sure that all your packages are using the same version of `@urql/core` so if after that `yarn list --pattern urql` still shows you two of those, then run `npx yarn-deduplicate`.

Similarly for `npm` there's the `npm dedupe` command.

Hope that helps 🙌
